scipy/special/_round.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Functions for adding two double precision numbers with rounding to
+ * infinity or rounding to negative infinity without using <fenv.h>.
+ *
+ * References
+ * ----------
+ * [1] Hida, Li, Bailey, "Library for Double-Double and Quad-Double
+ *     Arithmetic."
+ */
+#ifndef ROUND_H
+#define ROUND_H
+
+#include <numpy/npy_math.h>
+#include "_c99compat.h"
+
+
+double add_round_up(double a, double b)
+{
+    double s, e, v;
```

Comment:
https://gcc.gnu.org/wiki/FloatingPointMath --- gcc claims to do it as expected on SSE2, but not necessarily with x87 fpu
https://randomascii.wordpress.com/2012/03/21/intermediate-floating-point-precision/